### PR TITLE
Insert padding for conv and pooling ops before bit-width extend operations of operands

### PR DESCRIPTION
**Pull request by aviator19941**
Moves the padding for conv and pooling kernels before extsi operations of operands.
Fixes https://github.com/nod-ai/SHARK-Turbine/issues/750

### CODE REVIEW

File: sharktank/sharktank/kernels/pooling_nchw_sum.py [hunk 1]
```diff
@@ -19,46 +19,56 @@
...
+    signature = "pooling_nchw_sum(Tensor a, int[] weights_size, int[] strides, int[] padding, int[] dilations) -> (Tensor)"
 
     def select(self, ksel: KernelSelection):
-        input_desc = ksel.arg_tensor(0)  # Shape [b, ] m, k
-        inputs_pad_desc = ksel.arg_tensor(1)
-        weights_size_desc = ksel.attr_list_int(2)  # Shape [2]
-        strides_desc = ksel.attr_list_int(3)  # Shape [2]
-        padding_desc = ksel.attr_list_int(4)  # Shape [2]
-        dilations_desc = ksel.attr_list_int(5)  # Shape [2]
+        inputs_pad_desc = ksel.arg_tensor(0)
```
Comment (stellaraccident) [inline sharktank/sharktank/kernels/pooling_nchw_sum.py:25]: At this point, I think these are just "inputs" and we define (in op comments) that it operates on explicitly padded values

File: sharktank/sharktank/kernels/pooling_nchw_sum.py [hunk 2]
```diff
@@ -19,46 +19,56 @@
...
-        inputs_pad = kb.arg_value(1)
+        inputs_pad = kb.arg_value(0)
         inputs_pad_tensor_type = RankedTensorType(inputs_pad.type)
-        weights = ksel.arg_descs[2].v
-        strides = ksel.arg_descs[3].v
-        padding = ksel.arg_descs[4].v
-        dilations = ksel.arg_descs[5].v
+        weights = ksel.arg_descs[1].v
+        strides = ksel.arg_descs[2].v
+        padding = ksel.arg_descs[3].v
```
Comment (stellaraccident) [inline sharktank/sharktank/kernels/pooling_nchw_sum.py:58]: Do you use these paddings anymore in the generated kernel?

Comment (aviator19941) [reply sharktank/sharktank/kernels/pooling_nchw_sum.py:58]: nope, good catch, will remove those

File: sharktank/sharktank/ops/qconv_impls.py [hunk 3]
```diff
@@ -112,8 +112,11 @@ def qconv2d_tensor_scaled_integer(
     stride = _expand_int_to_2_tuple(stride)
     padding = _expand_int_to_2_tuple(padding)
     dilation = _expand_int_to_2_tuple(dilation)
+    extended_list = [item for item in padding for _ in range(2)]
```
Comment (stellaraccident) [inline sharktank/sharktank/ops/qconv_impls.py:115]: Please name `extended_padding_list` (or something that indicates this is padding related)

File: sharktank/sharktank/ops/qconv_impls.py [hunk 4]
```diff
@@ -211,7 +216,7 @@ def _invoke_int32_conv2d(
             # may want to support an optional bias through the kernel.
             bias = torch.zeros((weight.shape[1],), dtype=input.dtype)
         y_qs = kernels.conv_2d_nchw_fchw(
-            input,
+            padded_input,
```
Comment (stellaraccident) [inline sharktank/sharktank/ops/qconv_impls.py:219]: Can't comment on unchanged lines. Please remove the `padding` function arg and remove from the FP conv2d call below.

File: sharktank/sharktank/ops/qconv_impls.py [hunk 5]
```diff
@@ -233,14 +238,14 @@ def _invoke_int32_conv2d(
...
-    input, kernel_size, stride, padding, dilation, *, accum_dtype
+    input, padded_input, kernel_size, stride, padding, dilation, *, accum_dtype
 ):
     """Invokes either a custom integer pooling sum or the built-in fp avg_pool2d
     kernel.
     """
     if debugging.flags.use_custom_int_conv_kernel:
         output = kernels.pooling_nchw_sum(
-            input,
+            padded_input,
```
Comment (stellaraccident) [inline sharktank/sharktank/ops/qconv_impls.py:248]: Same: please remove the padding function arg and the padding=padding from the fp call below.

**Review by stellaraccident — APPROVED** [review]
A couple of nits and two bug fixes requested. You can land after making those changes.

File: sharktank/sharktank/ops/qconv_impls.py (hunk 4 above)
Comment (aviator19941) [reply sharktank/sharktank/ops/qconv_impls.py]: done

File: sharktank/sharktank/ops/qconv_impls.py (hunk 5 above)
Comment (aviator19941) [reply sharktank/sharktank/ops/qconv_impls.py]: done

**Review by stellaraccident — APPROVED** [review]
Still a nit: can you remove the padding from both _invoke_int32_pooling_sum and _invoke_int32_conv2d functions? It is unused.

Otherwise, lgtm.
